client/components/experiment/index.tsx

```diff
@@ -22,7 +23,21 @@ export { default as LoadingVariations } from './loading-variations';
  * @param props The properties that describe the experiment
  */
 export const Experiment: FunctionComponent< ExperimentProps > = ( props ) => {
-	const { isLoading: loading, variation, children } = props;
+	const { isLoading: loading, variation, children, name: experimentName } = props;
+	const [ eventFired, setEventFired ] = useState< boolean >( false );
+	useEffect( () => {
+		setEventFired( true );
+	}, [] );
```

Comment:
It's probably worth adding an inline comment explaining that this is in `useEffect()` to avoid firing multiple events.